src/server/scripts/EasternKingdoms/BlackrockMountain/BlackwingLair/boss_broodlord_lashlayer.cpp

```diff
@@ -113,7 +122,86 @@ class boss_broodlord : public CreatureScript
     }
 };
 
+class go_suppression_device : public GameObjectScript
+{
+    public:
+        go_suppression_device() : GameObjectScript("go_suppression_device") { }
+
+        struct go_suppression_deviceAI : public GameObjectAI
+        {
+            go_suppression_deviceAI(GameObject* go) : GameObjectAI(go), _instance(go->GetInstanceScript())
+            {
+                _events.ScheduleEvent(EVENT_SUPPRESSION_AURA, Seconds(6));
+                _events.ScheduleEvent(EVENT_SUPPRESSION_ANIM, Seconds(2));
+            }
+
+            void UpdateAI(uint32 diff) override
+            {
+                if (_instance->GetBossState(DATA_VAELASTRAZ_THE_CORRUPT) != DONE)
+                    return;
+
+                if (_instance->GetBossState(DATA_VAELASTRAZ_THE_CORRUPT) == DONE)
+                {
+                    if (me->GetGoState() == GO_STATE_DESTROYED)
+                        me->SetLootState(GO_READY); // NOT doing this will despawn the gameobject after using Disarm Trap
+                }
```

Comment:
```suggestion
                if (me->GetGoState() == GO_STATE_DESTROYED)
                    me->SetLootState(GO_READY); // NOT doing this will despawn the gameobject after using Disarm Trap
```

Reply:
thank you :)